common/lib/xmodule/xmodule/xml_module.py

```diff
@@ -512,7 +512,7 @@ def non_editable_metadata_fields(self):
         return non_editable_fields
 
 
-class XmlDescriptor(XmlParserMixin, XModuleDescriptor):  # pylint: disable=abstract-method
+class XmlMixin(XmlParserMixin):
```

Comment:
We want all these methods for `ProblemBlock` without the `XModuleDescriptor` so moved them to a mixin.